test/connectivity.jl

```diff
@@ -256,6 +256,14 @@
     @test @inferred(!isgraphical([1, 1, 1]))
     @test @inferred(isgraphical([2, 2, 2]))
     @test @inferred(isgraphical(fill(3, 10)))
+
+    # Test directed simple graphicality
+    for i in 1:3
+      sdg = SimpleDiGraph(10, 90)
+      @test @inferred(isdigraphical(indegree(sdg), outdegree(sdg)))
+    end
+    @test @inferred(!isdigraphical([1,1,1], [1,1,0]))
```

Comment:
The `@inferred` macro checks  the first call, and in this case it checks whether `!` infers correctly, so you should put that method call out of the macro scope.
(also in the test of `is_graphical` is not correct).
```suggestion
    @test !@inferred(isdigraphical([1,1,1], [1,1,0]))
```